knowledge-base/blog-grpc.md

```diff
@@ -0,0 +1,56 @@
+
+![The Fiddler Everywhere ninja to the resque](../images/temp/fe-ninja.png)
+
+After the Fiddler's team recently introduced the support for HTTP/2 and TLS 1.3, it was natural to go ahead with the implementation for gRPC support. With Fiddler Everywhere version 4.2.0 and above, gRPC support is a reality, and you can start capturing and inspecting gRPC traffic out-of-the-box. Ok, not precisely out-of-the-box - because gRPC relays on HTTP/2, you must ensure that your Fiddler Everywhere application has HTTP/2 capturing enabled through **Settings > Connections > Enable HTTP/2 Support (BETA)**.
+
+![Enabling HTTP/2 capturing in Fiddler Everywhere](../images/temp/fe-enable-http2.png)
+
+That's all you need from Fiddler Everywhere's side! 
+
+The next step is to tell your gRPC client application to respect the Fiddler's proxy for its HTTP and HTTPS requests. The specific proxy configuration entirely depends on the used client's application technology (for example, different approaches if it is a browser, a Python app, a GO app, a Java app, and so on). [Learn more on how to set Fiddler Everywhere as a proxy for your technology here ...](https://docs.telerik.com/fiddler-everywhere/knowledge-base/capturing-grpc-traffic)
+
+## gRPC Inspection
+
+Upon capturing a gRPC session, Fiddler Everywhere shows opened and closed gRPC channels in the Live Traffic list. A green badge in the ID column (by default, the first column on the left) indicates an active gRPC channel, while a red bad indicates that the gRPC channel is closed. Fiddler Everywhere supports all gRPC life cycles and will successfully capture unary RPC, client streaming RPC, server streaming RPC, and bidirectional streaming RPC.
+
+![Captured gRPC sessions in the Live Traffic list of Fiddler](../images/temp/fe-grpc-sessions.png)
+
+You can select a session with a single click to observe the **Handshake** tab and its details. The **Handshake** tab contains information about the used HTTP headers and the specific gRPC traliers. The server sends trailers to the client after processing the request. A typical scenario is for trailers to contain information common to service methods.
+
+![Inspecting gRPC session through the Handshake tab](../images/temp/fe-grpc-handshake-tab.png)
+
+You can open the **Messages** tab by double-clicking on the selected session (or by manually selecting the tab) and inspect each streamed message in the gRPC channel. Each gRPC message uses [the Protobuf format](https://protobuf.dev/overview/), which is in unreadable form and received as binary. You can inspect the binary message from the **Message** inspector. Fiddler Everywhere also provides instruments (**Decode value** and **HEX** inspector) to partially visualize text content that could be part of the Protobuf message. It is important to note that this will always be mutated content, not the original Protobuf message. Only the creator & owner of the .proto scheme can fully decode a Protobuf message.
```

Comment:
```suggestion
You can open the **Messages** tab by double-clicking on the selected session (or by manually selecting the tab) and inspect each streamed message in the gRPC channel. Each gRPC message uses [the Protobuf format](https://protobuf.dev/overview/), which is in unreadable form and received as binary. You can inspect the binary message from the **Message** inspector. Fiddler Everywhere also provides instruments (**Decode value** and **HEX** inspector) to partially visualize text content that could be part of the Protobuf message. It is important to note that this will always be the raw Protobuf content that was captured over the wire, not the decoded values of the message fields. Only the creator & owner of the .proto scheme can fully decode a Protobuf message. In a future release we will be adding the possibility for providing a .proto schema to Fiddler Everywhere for automatically decoding the Protobuf messages.
```